packages/angular/src/builders/library/build/build.impl.ts

```diff
@@ -0,0 +1,322 @@
+      ) {
+        const libPackageJson = readJsonFile(
+          join(context.workspaceRoot, depNode.data.root, 'package.json')
+        );
+
+        return {
+          scope: libPackageJson.name, // i.e. @wrkspace/mylib
+          outputPath: `dist/${depNode.data.root}`,
+          node: depNode
+        };
+      } else {
+        return null;
+      }
+    })
+    .filter(x => !!x);
+}
+
+/**
+ * Updates the peerDependencies section in the `dist/lib/xyz/package.json` with
+ * the proper dependency and version
+ */
+function updatePackageJsonDependencies(
+  context: BuilderContext,
+  libDependencies: DependentLibraryNode[]
+) {
+  const targetProject = context.target.project;
+
+  const projGraph = createProjectGraph();
+  const targetProjNode = projGraph.nodes[targetProject];
+
+  let distLibOutputPath = `dist/${targetProjNode.data.root}`;
+
+  // if we have dependencies, update the `dependencies` section of the package.json
+  const jsonOutputFile = `${distLibOutputPath}/package.json`;
+  if (libDependencies && libDependencies.length > 0) {
+    const outputJson = readJsonFile(jsonOutputFile);
+
+    outputJson.dependencies = outputJson.dependencies || {};
+
+    libDependencies.forEach(entry => {
+      if (!outputJson.dependencies[entry.scope]) {
+        // read the lib version (should we read the one from the dist?)
+        const packageJsonPath = join(
+          context.workspaceRoot,
+          entry.node.data.root,
+          'package.json'
+        );
+        const depNodePackageJson = readJsonFile(packageJsonPath);
+
+        outputJson.dependencies[entry.scope] = depNodePackageJson.version;
+      }
+    });
+
+    writeJsonFile(jsonOutputFile, outputJson);
+  }
+}
+
+async function scheduleLibraryBuilds(
+  options: BuildAngularLibraryBuilderOptions & JsonObject,
+  context: BuilderContext,
+  builds: DependentLibraryNode[]
+) {
+  let allBuildsSuccess = true;
+
+  for (const b of builds) {
+    // schedule a build
+    const buildRun = await context.scheduleTarget(
+      {
+        project: b.node.name,
+        target: 'build'
+      },
+      {
+        // overwrite the withDeps prop as we probably want to pass that along to
+        // child builds as well. otherwise it would be weird
+        withDeps: options.withDeps
+      }
+    );
+
+    // wait for the result
+    const result = await buildRun.result;
+
+    if (result.success === false) {
+      allBuildsSuccess = false;
+
+      // stop building other libs or 🤔 could we continue in some cases?
+      break;
+    }
+  }
+
+  return { success: allBuildsSuccess };
+}
+
+export function run(
+  options: BuildAngularLibraryBuilderOptions & JsonObject,
+  context: BuilderContext
+): Observable<BuilderOutput> {
+  return of({
+    options,
+    context,
+    dependencies: calculateLibraryDependencies(context),
+    result: { success: true }
+  }).pipe(
+    // verify param combination
+    switchMap(buildProps => {
+      if (
```

Comment:
We can do this higher like on line 234.

```ts
run(options) {
  if (options.withDeps && options.watch) {
    throw new Error('Using --withDeps in combination with --watch is not supported');
  }
}
```

Not completely sure, but I believe throwing an error is more idiomatic for this. Emitting `success: false` also works though.